src/shell/environmentshell.ts

```diff
@@ -1,7 +1,23 @@
 const spawn = require('cross-spawn')
 
 export default class EnvironmentShell {
-  static netstatCmd(): string {
-    return spawn.sync("netstat -vanp tcp | grep '3306' | grep -v 'TIME_WAIT' | grep -v 'CLOSE_WAIT' | grep -v 'FIN_WAIT'")
+  static isPortAvailable(port: number): boolean {
+    const result = EnvironmentShell.netstatCmd(port)
+    return result.status !== 0
+  }
+
+  static netstatCmd(port: number): any {
+    const portText = EnvironmentShell.isLinuxOs()
+      ? `:${port} `
+      : `.${port} `
+
+    // @TODO: is there any danger in concatenating the command like this? Do we need to escape it somehow?
```

Comment:
@josecanhelp is this a problem? In PHP we usually use `escapeshellarg` when adding arguments to command like this. Didn't find references to things like that in NodeJS, so I wasn't sure.